plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -503,6 +503,12 @@ func GetResourceRequest(pod *v1.Pod) *schedulercache.Resource {
 						result.SetExtended(rName, value)
 					}
 				}
+				if v1helper.IsHugePageResourceName(rName) {
```

Comment:
It looks to me that all the changes to the scheduler code could have been implemented with ExtendedResources instead of a new HugePageResource type. Why is it necessary to have this new resource type?

Reply:
@bsalamat my understanding is that extended resources are not in the default kubernetes resource namespace.

see: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/helper/helpers.go#L121

the intention is that hugepages are a natively understood resource upon graduation and are in the kubernetes resource namespace.

Reply:
From scheduler's point of view Extended Resources, OIR, or HugePages are very similar and the logic to handle them should be the same. That's why I am not happy with duplicating the code for HugePages in scheduler.
Given that we are very close to the code freeze for 1.8, I am fine with the changes if we coalesce the code for these resources in a follow-on.